resmem.c

```diff
@@ -50,37 +52,295 @@ static inline int get_info_infile(char *fname, char *res, void *out)
 	return 0;
 }
 
+/* read information from a cgroup file.
+ */
+static inline size_t cgmemread(char *cg, char *file)
+{
+	char buf[MEMBUF_128];
+	char fn[FNAMELEN];
+
+	snprintf(fn, FNAMELEN, "%s/%s%s/%s", DEFAULTCGFS,
+		MEMCGNAME, cg, file);
+
+	if (file_to_buf(fn, buf, MEMBUF_128) == -1)
+		return 0;
+
+	return (strtoul(buf, NULL, 10) / 1024);
+}
+
+/* read memory limit from cgorup file
+ */
+static size_t cgmemlimit(char *cg, char *f)
+{
+	char *copy;
+	char *dir;
+	size_t mem = 0, retmem = 0;
+
+	/* read memory limit for cgroup */
+	if ((retmem = cgmemread(cg, f)) == -1) {
+		return 0;
+	}
+
+	copy = strdup(cg);
```

Comment:
Probably want to check memory allocation error here.

Reply:
will update the code for check.